pkg/virt-controller/watch/clone/clone.go

```diff
@@ -355,12 +355,201 @@ func (ctrl *VMCloneController) updateStatus(origClone *clonev1alpha1.VirtualMach
+	}
+	snapshot := obj.(*snapshotv1alpha1.VirtualMachineSnapshot)
+	syncInfo.logger.Infof("found snapshot %s for clone %s", snapshot.Name, vmClone.Name)
+
+	if !virtsnapshot.VmSnapshotReady(snapshot) {
+		syncInfo.logger.V(defaultVerbosityLevel).Infof("snapshot %s for clone %s is not ready to use yet", snapshot.Name, vmClone.Name)
+		return snapshot, syncInfo
+	}
+
+	ctrl.logAndRecord(vmClone, SnapshotReady, fmt.Sprintf("snapshot %s for clone %s is ready to use", snapshot.Name, vmClone.Name))
+	syncInfo.snapshotReady = true
+
+	return snapshot, syncInfo
+}
+
+// This method assumes the snapshot exists. If it doesn't - syncInfo is updated accordingly.
+func (ctrl *VMCloneController) getSnapshot(vmClone *clonev1alpha1.VirtualMachineClone, sourceNamespace string, syncInfo syncInfoType) (*snapshotv1alpha1.VirtualMachineSnapshot, syncInfoType) {
+	obj, exists, err := ctrl.snapshotInformer.GetStore().GetByKey(getKey(*vmClone.Status.SnapshotName, sourceNamespace))
+	if !exists {
+		// At this point the snapshot is already created. If it doesn't exist it means that it's deleted for some
+		// reason and the clone should fail
+		syncInfo.isCloneFailing = true
+		syncInfo.failEvent = SnapshotDeleted
+		syncInfo.failReason = fmt.Sprintf("snapshot %s does not exist anymore", *vmClone.Status.SnapshotName)
+		return nil, syncInfo
+	}
+	if err != nil {
+		return nil, addErrorToSyncInfo(syncInfo, fmt.Errorf("error getting snapshot %s from cache for clone %s: %v", *vmClone.Status.SnapshotName, vmClone.Name, err))
+	}
+	snapshot := obj.(*snapshotv1alpha1.VirtualMachineSnapshot)
+
+	return snapshot, syncInfo
+}
+
+func (ctrl *VMCloneController) createRestoreFromVm(vmClone *clonev1alpha1.VirtualMachineClone, vm *k6tv1.VirtualMachine, snapshotName string, syncInfo syncInfoType) syncInfoType {
+	patches := generatePatches(vm, &vmClone.Spec)
+	restore := generateRestore(vmClone.Spec.Target, vm.Name, vmClone.Namespace, vmClone.Name, snapshotName, vmClone.UID, patches)
+	syncInfo.logger.Infof("creating restore %s for clone %s", restore.Name, vmClone.Name)
+
+	restore, syncInfo.err = ctrl.client.VirtualMachineRestore(restore.Namespace).Create(context.Background(), restore, v1.CreateOptions{})
+	if syncInfo.err != nil {
+		return addErrorToSyncInfo(syncInfo, fmt.Errorf("failed creating restore %s for clone %s: %v", restore.Name, vmClone.Name, syncInfo.err))
+	}
+
+	ctrl.logAndRecord(vmClone, RestoreCreated, fmt.Sprintf("created restore %s for clone %s", restore.Name, vmClone.Name))
+	syncInfo.restoreName = restore.Name
+
+	syncInfo.logger.V(defaultVerbosityLevel).Infof("restore %s was just created, reenqueuing to let snapshot time to finish", restore.Name)
+	return syncInfo
+}
+
+func (ctrl *VMCloneController) verifyRestoreReady(vmClone *clonev1alpha1.VirtualMachineClone, sourceNamespace string, syncInfo syncInfoType) syncInfoType {
+	obj, exists, err := ctrl.restoreInformer.GetStore().GetByKey(getKey(*vmClone.Status.RestoreName, sourceNamespace))
+	if !exists {
```

Comment:
I see a mixed coding style here, in previous places you are using the following pattern:
```
obj, exists, err := ctrl.someInformer.GetStore()...
if ! exists {
}
if err != nil {
}
snapshot := obj.(*...  // no newline after if block
```
In the current place you are using 
```
obj, exists, err := ctrl.someInformer.GetStore()...
if ! exists {
}
else if err != nil {
}

snapshot := obj.(*...  // newline after if block
```
I would advise to choose one of them for consistency matters

Reply:
I agree! Thanks for catching that